src/main/java/org/osiam/ResourceServer.java

```diff
@@ -0,0 +1,220 @@
+        Flyway flyway = new Flyway();
+        flyway.setBaselineOnMigrate(true);
+        flyway.setDataSource(dataSource);
+        flyway.setLocations("db/migration/" + databaseVendor);
+        flyway.setTable("resource_server_schema_version");
+        MigrationVersion version = MigrationVersion.fromVersion("0");
+        flyway.setBaselineVersion(version);
+        return flyway;
+    }
+
+    @Bean
+    @DependsOn("flyway")
+    public EntityManagerFactory entityManagerFactory() throws SQLException {
+        HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        vendorAdapter.setGenerateDdl(false);
+        vendorAdapter.setShowSql(false);
+        vendorAdapter.setDatabasePlatform(databasePlatform);
+
+        Properties jpaProperties = new Properties();
+        jpaProperties.put("hibernate.hbm2ddl.auto", "validate");
+        jpaProperties.put("hibernate.ejb.naming_strategy", "org.hibernate.cfg.ImprovedNamingStrategy");
+
+        LocalContainerEntityManagerFactoryBean factory = new LocalContainerEntityManagerFactoryBean();
+        factory.setJpaVendorAdapter(vendorAdapter);
+        factory.setPackagesToScan("org.osiam.storage.entities");
+        factory.setDataSource(dataSource());
+        factory.setJpaProperties(jpaProperties);
+        factory.afterPropertiesSet();
+
+        return factory.getNativeEntityManagerFactory();
+    }
+
+    @Bean
+    public JpaTransactionManager transactionManager() throws SQLException {
+        JpaTransactionManager txManager = new JpaTransactionManager();
+        txManager.setEntityManagerFactory(entityManagerFactory());
+        return txManager;
+    }
+
+    @Bean
+    public MetricRegistry metricRegistry() {
+        return new MetricRegistry();
+    }
+
+    @Bean
+    public HibernateExceptionTranslator hibernateExceptionTranslator() {
+        return new HibernateExceptionTranslator();
+    }
+
+    @Bean
+    public ShaPasswordEncoder passwordEncoder() {
+        ShaPasswordEncoder passwordEncoder = new ShaPasswordEncoder(512);
+        passwordEncoder.setIterations(1000);
+        return passwordEncoder;
+    }
+
+    @Configuration
+    @EnableResourceServer
+    protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+
+        @Autowired
+        private AccessTokenValidationService accessTokenValidationService;
+
+        @Bean
+        public OAuth2AuthenticationEntryPoint entryPoint() {
+            OAuth2AuthenticationEntryPoint entryPoint = new OAuth2AuthenticationEntryPoint();
+            entryPoint.setRealmName("oauth2-resource-server");
+            return entryPoint;
+        }
+
+        @Override
+        public void configure(ResourceServerSecurityConfigurer resources) {
+            resources
+                    .resourceId("oauth2res")
+                    .tokenServices(accessTokenValidationService)
+                    .expressionHandler(new OAuth2WebSecurityExpressionHandler());
```

Comment:
I changed it from `DefaultWebSecurityExpressionHandler` to `OAuth2WebSecurityExpressionHandler` because you [commented on an outdated diff](https://github.com/osiam/resource-server/pull/50#discussion_r36991089).